index.js

```diff
@@ -58,7 +63,10 @@ module.exports = exports = function fetch(input, init = {}) {
       },
       (res) => {
         if (res.headers.location && isRedirectStatus(res.statusCode)) {
-          return process(res.headers.location)
+          const crossOriginRedirect =
+            request._url.origin !== new URL(res.headers.location).origin
```

Comment:
Make a function for this called `isSameOrigin()` that implements the check in https://html.spec.whatwg.org/multipage/browsers.html#same-origin, but without relying on `url.origin` as we intentionally don't implement that. We should test `url.protocol` and `url.host` directly.